tools/build_launch_agent.py

```diff
@@ -14,6 +14,11 @@
     && apt-get -qy autoremove \
     && apt-get clean && rm -r /var/lib/apt/lists/*
 
+# copy reqs and install for docker cache
+COPY ../requirements.txt /src/requirements.txt
```

Comment:
Has this been tested? Not sure if COPY works for this, from the [docs](https://docs.docker.com/engine/reference/builder/#copy):
> The `<src>` path must be inside the context of the build; you cannot `COPY ../something /something`, because the first step of a `docker build` is to send the context directory (and subdirectories) to the docker daemon.

Reply:
Yes it works, as long as you run the script from the root of the repo. The Dockerfile also already copies the whole repo into the container with COPY .. /src